async/pool.ts

```diff
@@ -40,7 +47,7 @@ export function pooledMap<T, R>(
         // fail the race, taking us to the catch block where all currently
         // executing jobs are allowed to finish and all rejections among them
         // can be reported together.
-        p.then((v) => writer.write(Promise.resolve(v))).catch(() => {});
+        writer.write(p);
```

Comment:
What happens if p rejects with AggregateError?

Reply:
Good point.

I think we should handle this with another Exception made specifically for this case, that could not be thrown in p?

However that would be a breaking change.

Maybe check AggregateError message in the condition too?

Reply:
@kt3k I improved the handling of `AggregateError` so `controller.error(e as unknown);` is called only when it is relevant.
No interference is `p` rejects with `AggregateError` anymore

Reply:
Thanks for updating! This looks much better 👍